### DISCUSSION TITLE
feature request: allow optional `reviver` for json parsing in `sessionStorage`

### DISCUSSION THREAD

**Discussion by chris-kruining** [remix-run/remix#4433] [Proposals]
I would like to serialize/deserialize some data models into the `sessisonStorage`

would it be possible to add optional argument `reviver` to https://github.com/remix-run/remix/blob/358d2200ceceec45084729f908360f896ae2f8de/packages/remix-node/sessions/fileStorage.ts#L32-L35
Which would be used in `JSON.parse` on https://github.com/remix-run/remix/blob/358d2200ceceec45084729f908360f896ae2f8de/packages/remix-node/sessions/fileStorage.ts#L64

This way I can control my serialization with JS' `toJSON` method and deserialization via the `reviver`

**Comment by chris-kruining** [comment]
another usecase might be super JSON for example

**Comment by MichaelDeBoey** [comment]
Related proposal: https://github.com/remix-run/remix/discussions/4543
